src/modal/modal-window.ts

```diff
@@ -76,4 +82,15 @@ export class NgbModalWindow implements OnInit,
     this._elWithFocus = null;
     this._renderer.setElementClass(document.body, 'modal-open', false);
   }
+
+  private _setAriaLabelledByToModal(): void {
+    const modalHeader: HTMLHeadingElement = this._elRef.nativeElement.getElementsByClassName('modal-title')[0];
+    if (modalHeader) {
+      const modalHeaderId: string = modalHeader.getAttribute('id') === null || modalHeader.getAttribute('id') === '' ?
+          `modal-header-${nextId++}` :
+          modalHeader.getAttribute('id');
+      this._renderer.setElementAttribute(modalHeader, 'id', modalHeaderId);
+      this._renderer.setElementAttribute(this._elRef.nativeElement, 'aria-labelledby', modalHeaderId);
+    }
+  }
```

Comment:
Hmm, I see what you are doing here but it is super, super hackish... And it doesn't have tests... 

I'm not saying that I've got a better solution atm but there is way too much low-level DOM manipulation going on here... As such I would like to think some more about it. 

In the meantime, could you please open an issue describing this missing functionality? Thnx!